File: api/src/MiniBookingSystem.Domain/Modules/Mentor/Entities/Mentor.cs
public class Mentor : BaseEntity
{
    public Guid UserId { get; set; }
    public string DisplayName { get; set; } = default!;
    public string Email { get; set; } = default!;
    public string? Bio { get; set; }
    public string? Specialization { get; set; }
    public int ExperienceYears { get; set; }
    public decimal BasePrice { get; set; }
    public string? AvatarUrl { get; set; }
    public bool IsActive { get; set; } = true;

    // Social links (all optional)
    public string? FacebookUrl { get; set; }
    public string? GithubUrl { get; set; }
    public string? LinkedInUrl { get; set; }
    public string? TelegramUrl { get; set; }
    public string? WebsiteUrl { get; set; }

    public ICollection<MentorSkill> Skills { get; set; } = new List<MentorSkill>();
    public ICollection<MentorSlot> Slots { get; set; } = new List<MentorSlot>();

    public void Initialize(
        Guid userId,
        string displayName,
        string email,
        string? bio,
        string? specialization,
        int experienceYears,
        decimal basePrice,
        string? avatarUrl,
        string? facebookUrl = null,
        string? githubUrl = null,
        string? linkedInUrl = null,
        string? telegramUrl = null,
        string? websiteUrl = null
    )
    {
        UserId = userId;
        DisplayName = displayName;
        Email = email;
        Bio = bio;
        Specialization = specialization;
        ExperienceYears = experienceYears;
        BasePrice = basePrice;
        AvatarUrl = avatarUrl;
        FacebookUrl = facebookUrl;
        GithubUrl = githubUrl;
        LinkedInUrl = linkedInUrl;
        TelegramUrl = telegramUrl;
        WebsiteUrl = websiteUrl;
    }

    public void Update(
        string? displayName,
        string? bio,
        string? specialization,
        int? experienceYears,
        decimal? basePrice,
        string? avatarUrl,
        string? facebookUrl = null,
        string? githubUrl = null,
        string? linkedInUrl = null,
        string? telegramUrl = null,
        string? websiteUrl = null
    )
    {
        if (displayName is not null)
            DisplayName = displayName;
        if (bio is not null)
            Bio = bio;
        if (specialization is not null)
            Specialization = specialization;
        if (experienceYears.HasValue)
            ExperienceYears = experienceYears.Value;
        if (basePrice.HasValue)
            BasePrice = basePrice.Value;
        if (avatarUrl is not null)
            AvatarUrl = avatarUrl;
        if (facebookUrl is not null)
            FacebookUrl = facebookUrl;
        if (githubUrl is not null)
            GithubUrl = githubUrl;
        if (linkedInUrl is not null)
            LinkedInUrl = linkedInUrl;
        if (telegramUrl is not null)
            TelegramUrl = telegramUrl;
        if (websiteUrl is not null)
            WebsiteUrl = websiteUrl;
    }
}

File: web/app/types/mentor/mentor.ts
/**
 * Frontend types for the Mentor domain.
 * Mirror BE DTOs exactly so no mapping is needed.
 */

export type SlotStatus = 1 | 2 | 3 | 4 | 5;

export const SLOT_STATUS = {
  Available: 1,
  FullyBooked: 2,
  Blocked: 3,
  Cancelled: 4,
  Completed: 5,
} as const;

export const SLOT_STATUS_LABEL: Record<number, string> = {
  1: "Available",
  2: "Fully Booked",
  3: "Blocked",
  4: "Cancelled",
  5: "Completed",
};

export interface Mentor {
  id: string;
  userId: string;
  displayName: string;
  email: string;
  phoneNumber?: string | null;
  bio: string | null;
  specialization: string | null;
  experienceYears: number;
  basePrice: number;
  avatarUrl: string | null;
  isActive: boolean;
  createdAt: string; // ISO string
}

export interface MentorSkill {
  id: string;
  skillName: string;
}

export interface MentorSlot {
  id: string;
  name: string;
  startTime: string; // ISO string
  endTime: string; // ISO string
  status: SlotStatus;
  price: number;
  description?: string | null;
  maxBookings: number;
  currentBookings: number;
}

export interface MentorDetail extends Mentor {
  skills: MentorSkill[];
  slots: MentorSlot[];
}
